tests/remote_resources/xtcp_ping_pong.py

```diff
@@ -0,0 +1,160 @@
+                                '  Sent message:  \n{}\n'.format(format_message(message, 40)) +
+                                '  Recv message:  \n{}\n'.format(format_message(returned_message, 40))
+                                )
+
+        except socket.timeout as err:
+            failures.append('FAILURE: Port timed out\n' +
+                            '  Local port:    {}\n'.format(sock.getsockname()[1]) +
+                            '  Remote port:   {}\n'.format(process_port)
+                            )
+
+        except socket.error as err:
+            failures.append('FAILURE: Connection failed\n' +
+                            '  Local port:    {}\n'.format(sock.getsockname()[1]) +
+                            '  Remote port:   {}\n'.format(process_port) +
+                            '  Error message: {}\n'.format(err)
+                            )
+
+            # Reconnect
+            (failures, sock) = attempt_connect(failures, process_port, sock)
+            if sock is None:
+                break
+
+        tests_to_perform -= 1
+    
+    if sock is not None:
+        sock.shutdown(socket.SHUT_RDWR)
+        sock.close()
+
+    return failures
+
+# Remote device is setup to exit() if the first character of a message is 'a'
+def kill_remote_device():
+    try:
+        sock = socket.socket(socket.AF_INET, args.protocol)
+        sock.settimeout(10) # seconds
+        sock.connect((args.ip, start_port))
+        sock.send('a')
+        sock.close()
+
+    except socket.error as err:
+        # Do nothing and wait for remote device to timeout
+        print 'ERROR: Could not kill remote device'
+
+def test_iteration():
+    # Each process handles one remote port 
+    pool = Pool(args.remote_processes * args.remote_ports)
+
+    # Each process on the xC device has a pool of ports it can
+    # read from, with each pool spaced 10 apart
+    port_pool = [ j + (10*i) for i in range(0, args.remote_processes) for j in range(args.start_port, args.start_port + args.remote_ports)]
+    pool.map_async(process_test, port_pool, 1, callback=print_errors).get(9999999)
+
+    pool.close()
+    pool.join()
+    pool.terminate()
+
+if __name__ == '__main__':
+    ip = "10.0.102.198" # Device given this reserved IP by the office's DHCP server
+    start_port = 15533
+    packets = 10000
+    packet_size = 1460 # Safe maximum packet size that is not split
+    delay_between_packets = 0.01
+    remote_processes = 4
+    remote_ports = 1
+    protocol = "TCP"
+
+    parser = argparse.ArgumentParser(description='TCP/UDP tester')
+    parser.add_argument('--remote-processes', default=remote_processes, type=int,
+      help="Remote processes to connect to (default='%s')" % remote_processes)
+    parser.add_argument('--remote-ports', default=remote_ports, type=int,
+      help="Remote ports to connect to (default='%s')" % remote_ports)
+    parser.add_argument('--ip', default=ip, type=str,
```

Comment:
I'd be tempted to make arguments such as the IP address mandatory and not provide defaults, to avoid the confusion later on when something stops working because the defaults were being relied on.